### PR TITLE
BluetoothAudio: change singleton implementation to Create/Destroy bas…

### PR DESCRIPTION
**Pull request by HaseenaSainul**
…ed on requirement

### CODE REVIEW

**Review by sebaszm — CHANGES_REQUESTED** [review]
This is still holding a central instance and will fail for init, init, deinit, deinit scenario (ie. multiple clients). It either needs to be reference counted or keeping of the instance needs to be offloaded to the client (ie the API passing a handle).

**Comment by HaseenaSainul** [comment]
> This is still holding a central instance and will fail for init, init, deinit, deinit scenario (ie. multiple clients). It either needs to be reference counted or keeping of the instance needs to be offloaded to the client (ie the API passing a handle).

updated the code

File: Source/bluetoothaudiosink/BluetoothAudioSink.cpp [hunk 1]
```diff
@@ -284,15 +278,24 @@ namespace BluetoothAudioSinkClient {
...
-            return (*instance);
+            ASSERT(_instance != nullptr);
+            return (*_instance);
         }
-        static void Dispose()
+        static void Create()
         {
-            ASSERT(_singleton != nullptr);
-            delete _singleton;
+            if (_instance == nullptr) {
```
Comment (sebaszm) [inline Source/bluetoothaudiosink/BluetoothAudioSink.cpp:286]: Doesnt this need a lock?

Comment (HaseenaSainul) [reply Source/bluetoothaudiosink/BluetoothAudioSink.cpp:287]: yes, added. Thanks for the comments :)

**Review by sebaszm — DISMISSED** [review]

**Review by pwielders — APPROVED** [review]

**Pull request merged**